From c8c7cc1157bfff98572a229bfcb90c754b9ab881 Mon Sep 17 00:00:00 2001
From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date: Wed, 13 Aug 2014 20:49:52 -0700
Subject: [PATCH 151/213] rbd: fix error return code in rbd_dev_device_setup()
References: fate#318918
Git-commit: 255939e783d8f45f8c58487dfc18957c44ea9871
Patch-mainline: v3.17-rc5

Fix to return -ENOMEM from the workqueue alloc error handling
case instead of 0, as done elsewhere in this function.

Reviewed-by: Alex Elder <elder@linaro.org>
Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Acked-by: David Disseldorp <ddiss@suse.de>

---
 drivers/block/rbd.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/block/rbd.c b/drivers/block/rbd.c
index 9562eee..e23c9b9 100644
--- a/drivers/block/rbd.c
+++ b/drivers/block/rbd.c
@@ -5138,8 +5138,10 @@ static int rbd_dev_device_setup(struct rbd_device *rbd_dev)
 	set_disk_ro(rbd_dev->disk, rbd_dev->mapping.read_only);
 
 	rbd_dev->rq_wq = alloc_workqueue("%s", 0, 0, rbd_dev->disk->disk_name);
-	if (!rbd_dev->rq_wq)
+	if (!rbd_dev->rq_wq) {
+		ret = -ENOMEM;
 		goto err_out_mapping;
+	}
 
 	ret = rbd_bus_add_dev(rbd_dev);
 	if (ret)
-- 
2.1.4

